com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java

```diff
@@ -157,9 +159,23 @@ private void removeOldOrphanedFiles(Date expirationTime) {
     }
 
     private void removeActivitiHistoricData(Date expirationTime) {
+        Set<String> historicProcessIds = new HashSet<String>();
         activitiFacade.getHistoricProcessInstancesFinishedAndStartedBefore(expirationTime)
             .stream()
-            .forEach(historicProcessInstance -> activitiFacade.deleteHistoricProcessInstance(historicProcessInstance.getId()));
+            .forEach(historicProcessInstance -> addProcessAndSubProcesses(historicProcessIds, historicProcessInstance.getId()));
+
+        removeHistoricProcesses(historicProcessIds);
+    }
+
+    private void addProcessAndSubProcesses(Set<String> historicProcessIds, String historicProcessId) {
+        historicProcessIds.add(historicProcessId);
+        List<String> subProcessIds = activitiFacade.getHistoricSubProcessIds(historicProcessId);
+        historicProcessIds.addAll(subProcessIds);
+    }
+
+    private void removeHistoricProcesses(Set<String> historicProcessIds) {
+        historicProcessIds.stream()
+            .forEach(historicProcessId -> activitiFacade.deleteHistoricProcessInstance(historicProcessId));

```

Comment:
Again, this can lead to some exceptions with the Activiti. Could we catch and log the exceptions here?

Reply:
tryToDeleteHistoricProcessInstance would execute safely and log any errors should they occur.